tsdb/engine/wal/wal.go

```diff
@@ -725,8 +720,17 @@ type Partition struct {
 		OpenSegmentFile    func(name string, flag int, perm os.FileMode) (file *os.File, err error)
 		Rename             func(oldpath, newpath string) error
 	}
+
+	// buffers for reading and writing compressed blocks
+	// We constrain blocks so that we can read and write into a partition
+	// without allocating
+	buf [partitionBufLen]byte
+	// snappy buffer calculation taken from snappy.EncodedLen
+	snappybuf [32 + partitionBufLen + partitionBufLen/6]byte
```

Comment:
It feels weird to inline over 32KB into a struct. Also, it seems like the buffer size may be something we want to adjust in a config file. Can you move the allocation of this to the constructor instead?